lightning/src/ln/peer_handler.rs

```diff
@@ -1028,7 +1033,8 @@ impl<Descriptor: SocketDescriptor, CM: Deref, RM: Deref, OM: Deref, L: Deref, CM
 
 								peer.their_node_id = Some(their_node_id);
 								insert_node_id!();
-								let features = InitFeatures::known();
+								let features = if self.message_handler.onion_message_handler.supports_onion_message_forwarding()
+									{ InitFeatures::known() } else { InitFeatures::known_without_onion_messages() };
```

Comment:
Could we set a member variable for this in the constructor? Would avoid repeating oursleves.